src/prefs/DevicePrefs.cpp

```diff
@@ -412,13 +539,33 @@ bool DevicePrefs::Commit()
 
    AudioIOLatencyDuration.Invalidate();
    AudioIOLatencyCorrection.Invalidate();
+
+   QualitySettings::DefaultSampleRate.Invalidate();
+
+   // The complex compound control may have value 'other' in which case the
+   // value in prefs comes from the second field.
+   if (mOtherDefaultSampleRate->IsEnabled())
+   {
+      QualitySettings::DefaultSampleRate.Write(mOtherDefaultSampleRateValue);
+      gPrefs->Flush();
+   }
+
+   if (mProject)
+   {
+      auto& projectRate = ProjectRate::Get(*mProject);
+      if (mOtherProjectSampleRate->IsEnabled())
+         projectRate.SetRate(mOtherProjectSampleRateValue);
+      else
+         projectRate.SetRate(mSampleRateValues[mProjectSampleRates->GetSelection()]);
```

Comment:
I noticed today (it escaped me in an earlier round of review) that the earlier change in DevicePrefs changed project rate immediately in the event handler and ignored dialog cancellation, but this now delays the side effect on the project until the OK button.

I think that's better, and consistent with how preferences are changed.  But is there any debate?

Reply:
Why there would be any debate?